src/PluginsSystem/Tools/Microsoft.Performance.Toolkit.Plugins.Cli/Processing/PluginArtifacts.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.Performance.Toolkit.Plugins.Cli.Processing
+{
+    /// <summary>
+    ///     Represents the artifacts of a plugin.
+    /// </summary>
+    /// <param name="SourceDirectoryFullPath">
+    ///     The full path to the directory containing the plugin binaries.
+    /// </param>
+    /// <param name="ManifestFileFullPath">
+    ///     The full path to the manifest file, if it's not in the source directory. Otherwise, <c>null</c>.
+    /// </param>
+    internal record PluginArtifacts(
```

Comment:
Instead of using null, I would say just have a boolean value `bool SourceDirectoryContainsManifestFile`

Reply:
Make it not nullable and just always have the path (regardless of whether or not it's in the source directory)

Reply:
Well if it's in the source directory, we won't know the path to the manifest file until we call `TryProcess()`. The assumption for now is that manifest can be placed anywhere in the source directory. TryProcess would search for the name of the manifest in the source directory. To make things easier, we can enforce that pluginManifest.json must be at the root level.

Reply:
Another approach is to instead make an interface `IManifestLocator` that has a single methof `string Locate()`. If it's within the source directory, we pass in a `SourceDirectoryManifestLocator`. Otherwise we pass in an implenetation that spits out the location provided on the command line

Reply:
Basically, delegate the work of knowing how to extract the manifest location to another class (that we can unit test in isolation)

Reply:
Yes